### PR TITLE
#8 remove user from  group

### PR DESCRIPTION
**Pull request by KhangZhengNg**
#8 

Allow supervisors to remove any user in the monitoring groups by long pressing the user and choose "remove" from the context menu. An Espresso UI test file is also created.

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/activity/WelcomePageActivity.java [hunk 1]
```diff
@@ -15,13 +16,10 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_welcome);
-        new Handler().postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                Intent i = new Intent(WelcomePageActivity.this, LoginPageActivity.class);
-                startActivity(i);
-                finish();
-            }
+        new Handler(Looper.getMainLooper()).postDelayed(() -> {
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/activity/WelcomePageActivity.java:19]: Should be done in a different PR since this is another issues' responsibility.

File: app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java [hunk 2]
```diff
@@ -52,4 +53,14 @@ private void loadAllUsersInGroup(Group group) {
         usersInGroup = userRepository.getDocumentsByListOfUId(group.getSupervisedUserId(), User.class);
     }
 
+    public void removeUserFromGroup(User userToBeRemoved) {
+        Group groupWithRemovedUser = getGroup().getValue();
+        Objects.requireNonNull(groupWithRemovedUser).getSupervisedUserId().remove(userToBeRemoved.getUId());
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java:58]: groupWithRemovedUser should be in an if (groupWithRemovedUser != null) with a throw exception, or a try-catch clause. This doesn't handle an exception.

**Review by ShawnGuerra-Bautista — CHANGES_REQUESTED** [review]
Few changes requested.
Otherwise, cool test.

File: app/src/main/java/com/example/bigowlapp/activity/WelcomePageActivity.java (hunk 1 above)
Comment (ShawnGuerra-Bautista) [reply app/src/main/java/com/example/bigowlapp/activity/WelcomePageActivity.java]: Finished it.

**Review by refatK — APPROVED** [review]
Code seems fine and the app ran fine for me as well. Removing a user from my monitored list worked as  expected, ui updates correctly, and so does the users in my monitoring list in the Database.

Good job adding a test as well!

File: app/src/androidTest/java/com/example/bigowlapp/activity/MonitoringGroupPageActivityTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    Group testMonitoringGroup;
+    List<User> testUserList;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        testUserList = new ArrayList<>();
+        List<String> supervisedUsersIDs = new ArrayList<>();
+        for (int i = 0; i < 6; i++) {
```
[the remark this answers is not in the document]
Comment (KhangZhengNg) [reply app/src/androidTest/java/com/example/bigowlapp/activity/MonitoringGroupPageActivityTest.java:68]: ok thank you!

File: app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java (hunk 2 above)
Comment (ShawnGuerra-Bautista) [reply app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java:58]: Not an issue; it's already handled. Forget about it.

Comment (KhangZhengNg) [reply app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java:58]: The method can only be called when a user can actually first see the monitoring group with a list of users, perform a long press click on any of the user, and click the "remove" button. Hence, no check is required for method removeUserFromGroup because the activity page and ViewModel already checked before a supervisor can even remove a user.

**Review by ShawnGuerra-Bautista — APPROVED** [review]
